dev/Code/CryEngine/CryPhysics/geoman.cpp

```diff
@@ -183,7 +183,13 @@ int CGeomManager::AddRefGeometry(phys_geometry* pgeom)
 
 int CGeomManager::UnregisterGeometry(phys_geometry* pgeom)
 {
-    WriteLock lock(m_lockGeoman);
+	WriteLock lock(m_lockGeoman);
+	return UnregisterGeometryLocked(pgeom);
```

Comment:
Moved the functionality into a private locked function so it can be called recursively